rfc/text-field.md

```diff
@@ -0,0 +1,170 @@
+---
+RFC: (PR number)
+Status: (Change to Proposed when it's ready for review)
+---
+
+# Title
+Text Searching
+
+## Abstract
+
+The existing search platform is extended with a new field type, ```TEXT``` is defined. The query language is extended to support locating keys with text fields based on term, wildcard, fuzzy and exact phrase matching.
+
+## Motivation
+
+Text searching is useful in many applications.
+
+## Terminology
+
+In the context of this RFC.
+
+| Term | Meaning |
+| --- | --- |
+| key | A Valkey key. Depending on the context of usage could be the text of the key name or the contents of the HASH or JSON key. |
+| field | A component of an index. Each field has a type and a path.|
+| index | A collection of fields and field-indexes. The object created by the ```FT.CREATE``` command. |
+| field-index | A data structure associated with a field that is intended to accelerate the operation of the search operators for this field type. |
+| character | An Unicode character. A character may occupy 1-4 bytes of a UTF-8 encoded string. |
+| word | A syntactic unit of text consisting of a vector of characters. A word is delimited by un-escaped punctuation and/or whitespace. |
+| token | same as a word |
+| text | A UTF-8 encoded string of bytes. |
+| stemming | A process of mapping similar words into a common base word. For example, the words _drives_, _drove_ and _driven_ would be replaced with the  _drive_. |
+| term | The output of stemming a word. |
+| prefix tree | A mapping from input string to an output object. Insert/delete operations are O(length(input)). Additional operations include the ability to iterate over the entries that share a common prefix in O(length(prefix)) time. |
+
+## Design considerations
+
+The text searching facility provides machinery that decomposes text fields into terms and field-indexes them.
+The query facility of the ```FT.SEARCH``` and ```FT.AGGREGATE``` commands is enhanced to select keys based on combinations of words, wildcard, fuzzy and phrase matching.
+
+### Tokenization process
+
+A tokenization process is applied to strings of text to produce a vector of terms.
+Tokensization is applied in two contexts. 
+First as part of the ingestion process for text fields of keys.
+Second to process query string words and phrases.
+
+The tokenization process has four steps.
+
+1. The text is tokenized, removing punctuation and redundant whitespace, generating a vector of words.
```

Comment:
Just listing a few questions. How would this handle the following?

1. Contractions (e.g. you're, I'm, They'll, etc) Will these be split into characters? URLs can be considered another form of a contraction
2. Emojis - Since we are not removing based on unicode, will emoji's survive tokenization?